### PR TITLE
Adding basic single plots to muon Analysis

### LINKED ISSUE #25738: Muon Analysis 2 multi plotting
Want to add plotting to the new muon GUI

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This adds basic single plotting to the new Muon Analysis GUI.

**To test:**
  * Ideally test in both workbench and mantidplot
  * Open Muon Analysis 2, when you load some data a plot window should pop up.
  * If you then do a fit on this data the fit should be added to the plot. 
  * Loading multiple runs should add all the runs to the graph and a fit on any of these graphs should then be added automatically. 
  * Check that plotting works correctly in frequency domain analysis
  * In mantidplot check the elemental analysis plotting still functions correctly.

Fixes #25738  

*This does not require release notes* because this adds functionality to a new interface which is having it's release notes done separately.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
if you do a fit. Then rerun the fit it will get added again to the plot (mantidplot), I also noticed that simultaneous fit only plots one. 

FDA crashes when you try to open it as it does not have fitting.
```
Traceback (most recent call last):
  File "c:\users\btr75544\work\dev\mantid\qt\applications\workbench\workbench\app\mainwindow.py", line 370, in <lambda>
    action.triggered.connect(lambda checked_py, script=script: self.launch_custom_python_gui(script))
  File "c:\users\btr75544\work\dev\mantid\qt\applications\workbench\workbench\app\mainwindow.py", line 347, in launch_custom_python_gui
    self.interface_executor.execute(open(filename).read(), filename)
  File "c:\users\btr75544\work\dev\mantid\qt\python\mantidqt\widgets\codeeditor\execution.py", line 172, in execute
    exec (code_obj, self.globals_ns, self.globals_ns)
  File "C:/Users/BTR75544/work/dev/mantid/scripts/Frequency_Domain_Analysis.py", line 28, in <module>
    muon_freq = FrequencyAnalysisGui()
  File "C:\Users\BTR75544\work\dev\mantid\scripts\Muon\GUI\FrequencyDomainAnalysis\frequency_domain_analysis_2.py", line 119, in __init__
    self.setup_on_recalulation_finished_notifer()
AttributeError: 'FrequencyAnalysisGui' object has no attribute 'setup_on_recalulation_finished_notifer'
```

File: scripts/test/Muon/home_plot_widget_test.py [hunk 1]
```diff
@@ -8,29 +8,113 @@
...
+    def test_handle_plot_type_changed_displays_a_warning_if_trying_to_plot_counts_on_a_pair(self):
+        self.context.group_pair_context.__getitem__.return_value = MuonPair('long', 'bwd', 'fwd')
+        self.view.get_selected.return_value = 'Counts'
+
+        self.presenter.handle_plot_type_changed()
+
+        self.model.plot.assert_not_called()
+        self.view.warning_popup.assert_called_once_with('Pair workspaces have no counts workspace')
+
+    def test_handle_plot_type_changed_calls_plot(self):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/home_plot_widget_test.py]: I don't think this test is right. As when I change the plot type I do not get an updated plot.

**Comment by AnthonyLim23** [comment]
The plotting is using the diff and not the fitted result (spec 2). To see this try a flat background with MUSR 62260

**Review by AnthonyLim23 — DISMISSED** [review]
@mantidproject/gatekeepers could we please merge this once the tests have passed

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew any idea why the `SANSILLReductionTest` has failed in every build?

**Comment by martyngigg** [comment]
I believe this was something that made it to master and is now currently disabled while it is investigated: https://github.com/mantidproject/mantid/commit/4e4ffc7e331d610b593d84f92b3206f275606a5b

**Comment by martyngigg** [comment]
@Matthew Andrew this has some conflicts now that the results tab changes in #25864 has been merged.

**Review by AnthonyLim23 — APPROVED** [review]
@mantidproject/gatekeepers could we please get this merged once the builds have passed

**Pull request merged**